### PR TITLE
MRG, DOC: Document 'ignore' on_defects value; make allowed on_missing values explicit

### PR DESCRIPTION
**Pull request by hoechenberger**
This is a follow-up to #9614, where @larsoner pointed out that the `on_defects` param should support an `'ignore'` value too. I've made the required adjustments, and also search-and-replaced `on_missing : str` -> `on_missing : 'raise' | 'warn' | 'ignore'` in the entire docdict.

### CODE REVIEW

File: mne/bem.py [hunk 1]
```diff
@@ -269,6 +269,8 @@ def _check_complete_surface(surf, copy=False, incomplete='raise', extra=''):
                        ', '.join(str(f) for f in fewer), extra))
         if incomplete == 'raise':
             raise RuntimeError(msg)
+        elif incomplete == 'ignore':
+            pass
         else:
             warn(msg)
```
Comment (larsoner) [inline mne/bem.py:275]: You should use `_on_missing` here

Comment (hoechenberger) [reply mne/bem.py]: Ok, done!

**Comment by larsoner** [comment]
Failures are real, maybe just an error class change?

**Comment by hoechenberger** [comment]
Arrrrrgh. Will have a look.

**Comment by hoechenberger** [comment]
@larsoner I think we should be good now. Thanks for your feedback!

**Comment by hoechenberger** [comment]
(Not sure why CI won't start up now?)

**Comment by hoechenberger** [comment]
(oh, webhook incident. well … we'll just have to wait until it's resolved, I suppose)

**Comment by hoechenberger** [comment]
Ok we're green!

**Comment by agramfort** [comment]
Thx @hoechenberger

**Pull request merged**